open-craft/edx-platform
# feat: adds Library Content button to Studio Unit page

Clicking the iframes the Authoring MFE's Library Content Picker.

![image](https://github.com/user-attachments/assets/c7a5597b-ac1e-4c75-a247-2e1ae4152c8a)

![image](https://github.com/user-attachments/assets/58cf0217-f6c2-441b-976a-c8259bfaf197)


### Testing instructions

1. Disable these waffle flags to access the legacy Studio Unit page:
   * `contentstore.new_studio_mfe.use_new_course_outline_page`
   * `contentstore.new_studio_mfe.use_new_unit_page`
1. Create a course and edit a Unit there.
2. Click the new Library Content to open the library content picker, and select a block to copy into the course.
3. Refresh the Unit page to see the newly added block (see concerns below).

### Author Concerns and Issues

With the Content Picker:
1. iframe event doesn't tell us what block ID was added to the Unit, only that we should close the picker. So we have to refresh the whole Unit page to see the added component.
4. Needs some style updates to fit within our iframe.

With this code:
1. Need to handle ESC and/or clicking outside the modal to cancel the operation.
2. Should add a placeholder element (like normal XBlocks do), so when the CP returns the new course block ID, we can just refresh that XBlock location.
